### ISSUE TITLE
Inter-version compatibility guide

### ISSUE THREAD

**Issue by yarkinwho** [blockscout/blockscout#7415]
I am trying to run the blockscout with separated indexer and webapp. 

The question is , what happen if indexer and webapp are in different version? This may happen during upgrade.

And if it do break things, what would be the best practice to upgrade the system with minimum disturbance? (at least better than bring everything down and upgrade one by one) 

Thanks.

**Comment by vbaranov** [comment]
@yarkinwho you should keep equal versions of web app and indexer in case of separate launches in order to prevent errors due to backward incompatibility.

In order to keep application availability during the update you may want to deploy Blockscout into k8s infrastructure, which provides the availability of the previous pod while the new one is deploying from the box, or develop a script that will provide blue-green deployment strategy.

**Comment by yarkinwho** [comment]
> 

Thanks for the quick reply. 

When we are talking about the blue-green deployment here, the problem is there are db schema changes, I don't think blue and green environment can work on the same database, right?

If this is the case, we will have to clone the db to make two separate working environment I suppose? 

Thanks.

**Comment by vbaranov** [comment]
Yes, this note is correct. However, it depends on the type of DB migration. If this is the creation of a new table or adding a new column or modifying quite a small table (adding index, modifying column type etc), it could be done on the same DB.
